src/main/java/org/team1540/robot2025/subsystems/vision/apriltag/AprilTagVision.java

```diff
@@ -0,0 +1,64 @@
+package org.team1540.robot2025.subsystems.vision.apriltag;
+
+import static org.team1540.robot2025.subsystems.vision.apriltag.AprilTagVisionConstants.*;
+
+import edu.wpi.first.wpilibj.Alert;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import org.littletonrobotics.junction.Logger;
+import org.team1540.robot2025.RobotState;
+import org.team1540.robot2025.subsystems.vision.apriltag.AprilTagVisionIO.PoseObservation;
+
+public class AprilTagVision extends SubsystemBase {
+    private final AprilTagVisionIO[] visionIOs;
+    private final AprilTagVisionIOInputsAutoLogged[] cameraInputs;
+
+    private final Alert[] disconnectedAlerts;
+
+    private AprilTagVision(AprilTagVisionIO... visionIOs) {
+        this.visionIOs = visionIOs;
+
+        this.cameraInputs = new AprilTagVisionIOInputsAutoLogged[visionIOs.length];
+        this.disconnectedAlerts = new Alert[visionIOs.length];
+        for (int i = 0; i < cameraInputs.length; i++) {
+            cameraInputs[i] = new AprilTagVisionIOInputsAutoLogged();
+            disconnectedAlerts[i] =
+                    new Alert("Oh dear, " + visionIOs[i].getName() + " is disconnected D;", Alert.AlertType.kWarning);
```

Comment:
Should probably have a more professional & readable error message (as in remove the "Oh dear" prefix, you can keep the emoticon if you want)